### PR TITLE
Add RotationDirection enum

### PR DESCRIPTION
**Pull request by BurtHarris**
Fixes: #214

### CODE REVIEW

**Comment by rhiju** [comment]
👍 from me

File: src/eterna/pose2D/RNALayout.ts [hunk 1]
```diff
@@ -16,7 +21,7 @@ export class RNATreeNode {
     public goX: number = 0;
     public goY: number = 0;
 
-    public rotationDirectionSign: number = 1; // 1 or -1 for counterclockwise or clockwise
+    public rotationDirectionSign: RotationDirection;
```
Comment (luxaritas) [inline src/eterna/pose2D/RNALayout.ts]:
```suggestion
    public rotationDirection: RotationDirection;
```

Super minor thing!

Comment (luxaritas) [reply src/eterna/pose2D/RNALayout.ts]: Oh, this will require modification wherever it's used, so you'll need to use VS Code's rename reference

Comment (BurtHarris) [reply src/eterna/pose2D/RNALayout.ts]: Sure, done.

Comment (BurtHarris) [reply src/eterna/pose2D/RNALayout.ts]: Done.

**Comment by luxaritas** [comment]
I’m pretty sure they are - trying to set it to not a value of the enum should fail

**Comment by BurtHarris** [comment]
Here's what I mean.    If you change RNALayout.ts line 24 to read
```
public rotationDirection: RotationDirection = 0;
```
Then Typescript generates no error messages, even though 0 isn't one of the listed members of `RotationDirection`.    Type-wise, it seems that any `number` is considered assignable to `RotationDirection`.

**Comment by BurtHarris** [comment]
The [type compatibility section](http://www.typescriptlang.org/docs/handbook/type-compatibility.html#enums) of the typescript docs explain the behavior:

>Enums are compatible with numbers, and numbers are compatible with enums. Enum values from different enum types are considered incompatible.

**Pull request merged**